coeus-impl/src/main/java/org/kuali/coeus/common/budget/impl/period/BudgetPeriodRule.java

```diff
@@ -340,6 +340,10 @@ private boolean isValidNewBudgetPeriod(Budget budget, BudgetPeriod newBudgetPeri
                         newBudgetPeriod.setBudgetPeriod(periodNum+1);
                     }
                     break;
+                }else if(newPeriodStartDate.compareTo(periodEndDate) <= 0) {
+                    dateCompareValue = "ERROR_NEW_PERIOD_START_BEFORE_PREVIOUS_END";
+                    saveErrors(dateCompareValue, errorMap, newPeriodDateParams);
+                    validNewBudgetPeriod = false;
```

Comment:
OK, I wrote this earlier but never actually submitted it.

This works last time I checked, though perhaps I didn't test this exact condition (pretty sure, but I'll do it again to be sure).  Since the periods are automatically sorted by the date, the new starting date 7/1/2014 would be compared to the first period ending at 6/30/2014 producing a 1 and fail the if condition.  Putting a break in will stop it from failing on the second period.  That I believe is needed (If nothing else than saving time from making unnecessary iterations through the loop).